src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```
@@ -62,11 +62,13 @@ object DepositHandler {
 
   def finalizeDeposit(mimeType: String)(implicit id: String): Try[Unit] = {
     log.info(s"[$id] Finalizing deposit")
+    baseDir = new File(SwordProps("basedir"))
```

We need some check that this directory actually exists and is read-accessible, so we fail fast if it isn't.

---

I added an method to check these things.

---

not sure what `val tempDir = new File(SwordProps("tempdir"), id)` represents, but shouldn't you do a similar check for this one?

---

Oh, wait, you only use it in `getBagDir(tempDir)` and there it turns out to be the `depositDir`. Is this a naming issue or what? `tempDir` does not really sound like the directory where the actual deposit is located...

---

The bag will be moved from the tempDir to the proper deposit directory.